packages/polkadart_scale_codec/test/test_types/bitvec_test.dart

```diff
@@ -2,123 +2,195 @@ import 'package:polkadart_scale_codec/polkadart_scale_codec.dart';
+      expect([1, 1, 0, 1, 1, 0, 0], equals(decoded));
     });
 
     test(
-        'On decoding \'double\' should throw exception \'AssertionException\' for BitVec<u8>',
+        'When \'0x1c5f\' is decoded then result \'[1, 1, 1, 1, 1, 0, 1]\' is produced',
         () {
-      final exceptionMessage =
-          'Source(data) -> `data` should be either String, List<int> or Uint8List.';
-      expect(
-          () => codec.decode(usageIndex, 0.0),
-          throwsA(predicate((e) =>
-              e is AssertionException && e.toString() == exceptionMessage)));
+      final decoded = Codec(types).decode(usageIndex, '0x1c5f');
+      expect([1, 1, 1, 1, 1, 0, 1], equals(decoded));
     });
-  });
 
-  //
-  // Exception on Encode at wrong value type: `BitVec<u8>`
-  group('Exception on Encode BitVec<u8>:', () {
     test(
-        'On encoding \'integer\' value should throw exception \'AssertionException\' for BitVec<u8>',
+        'When \'0x201b\' is decoded then result \'[1, 1, 0, 1, 1, 0, 0, 0]\' is produced',
         () {
-      final exceptionMessage =
-          'BitSequence can have bits of type List<int> only.';
-      expect(
-          () => codec.encode(usageIndex, 0),
-          throwsA(predicate((e) =>
-              e is AssertionException && e.toString() == exceptionMessage)));
+      final decoded = Codec(types).decode(usageIndex, '0x201b');
+      expect([1, 1, 0, 1, 1, 0, 0, 0], equals(decoded));
     });
 
     test(
-        'On encoding \'float\' value should throw exception \'AssertionException\' for BitVec<u8>',
+        'When \'0x20ff\' is decoded then result \'[1, 1, 1, 1, 1, 1, 1, 1]\' is produced',
         () {
-      final exceptionMessage =
-          'BitSequence can have bits of type List<int> only.';
-      expect(
-          () => codec.encode(usageIndex, 0.0),
-          throwsA(predicate((e) =>
-              e is AssertionException && e.toString() == exceptionMessage)));
+      final decoded = Codec(types).decode(usageIndex, '0x20ff');
+      expect([1, 1, 1, 1, 1, 1, 1, 1], equals(decoded));
     });
   });
+
+  {
+    // Creates the registry for parsing the types
+    final registry = TypeRegistry();
+
+    // specifying which type to use.
+    final usageIndex = registry.getIndex('BitVec<u8, bitvec::order::Lsb0>');
+
+    // fetching the parsed types from `Json` to `Type`
+    final types = registry.getTypes();
+    //
+    // Exception on Encode at (lowest - 1) and (highest + 1): `BitVec<u8, bitvec::order::Lsb0>`
+    group('Exception on Encode BitVec<u8, bitvec::order::Lsb0>:', () {
+      test(
+          'should throw exception \'UnexpectedCaseException\' when encoding [-1] which is 1 lesser than lowest acceptable.',
+          () {
+        expect(
+            () => Codec(types).encode(usageIndex, [-1]),
+            throwsA(predicate((e) =>
+                e is UnexpectedCaseException &&
+                e.toString() == 'Invalid byte, unable to encode [-1].')));
```

Comment:
We need a single failing condition for each expectation

Reply:
This also applies to all tests

Reply:
I want to be sure that the error is showing the correct error message and its not mistakenly edited in ay further PR. Thats why I prefer to keep the error messages check also.

Reply:
I agree that we should keep both checks, but I think it's important that we're clear on the error message that we display in our framework. We want developers to be sure of what's going wrong, therefore we need to be specific and as detailed as possible when something fails 
```suggestion
            throwsA(predicate((e) {
                expect(e, isA<UnexpectedCaseException>());
                expect(e.toString(), 'Invalid byte, unable to encode [-1].');
            }
```

In the above suggestion, it's clear why our test fails